package ru.demo.wms.exception;
/*
Класс PartNotFoundException является пользовательским исключением, 
созданным для обработки ситуаций, когда часть или компонент в системе управления складом не найден. 
Это исключение наследуется от RuntimeException, что позволяет его использовать 
для контроля исполнения программы без необходимости явного объявления в сигнатурах методов.

Конструкторы исключения:

Пустой конструктор PartNotFoundException(), который вызывает базовый 
конструктор RuntimeException без аргументов. Этот конструктор может быть использован, 
когда не требуется предоставлять дополнительную информацию об ошибке.
Конструктор с параметром PartNotFoundException(String message), который принимает строку,
содержащую сообщение об ошибке. Это сообщение затем может быть использовано для логирования, 
отладки или отображения пользователю, предоставляя детали о том, почему конкретная часть не была найдена.
Использование этого исключения в коде позволяет разработчикам более точно указать ошибку, 
связанную с отсутствием части, и обеспечить более чистую и понятную обработку ошибок в приложении. 
Например, при попытке получить доступ к детали по её идентификатору в базе данных, если такой детали нет,
можно сгенерировать PartNotFoundException, что является более информативным, 
чем общее исключение RuntimeException или NullPointerException.
*/
public class PartNotFoundException extends RuntimeException {

	private static final long serialVersionUID = 1L;

	public PartNotFoundException() {
		super();
	}
	
	public PartNotFoundException(String message) {
			super(message);
	}
}
